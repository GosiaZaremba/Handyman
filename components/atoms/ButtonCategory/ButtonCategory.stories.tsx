import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {ButtonCategory} from './ButtonCategory';
import {action} from '@storybook/addon-actions';
import {text} from '@storybook/addon-knobs';

storiesOf('atoms/ButtonCategory', module)
  .addDecorator(story => story())
  .add('default', () => (
    <ButtonCategory
      onPress={action('white button click')}
      category={require('../../../assets/png/05_icons_cleaning.png')}
      text={text('button text', 'Laundry')}
      isActive={true}
    />
  ));
