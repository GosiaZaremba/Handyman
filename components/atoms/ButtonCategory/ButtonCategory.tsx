import React from 'react';
import {Pressable, Image, View, ImageSourcePropType} from 'react-native';
import {TextSmall} from '../TextSmall/TextSmall';
import styles from './ButtonCategory.styles';
import Icon from 'react-native-vector-icons/AntDesign';

export type Props = {
  onPress?: () => void;
  category?: ImageSourcePropType | undefined;
  text: string;
  isActive: boolean;
  testID?: string;
  testOnly_pressed?: boolean;
};

export const ButtonCategory: React.FC<Props> = ({
  category,
  text,
  onPress,
  isActive,
  testID,
  testOnly_pressed,
}) => {
  return (
    <View style={styles.container}>
      <Pressable
        testID={testID}
        hitSlop={5}
        style={isActive ? styles.pressed : styles.pressable}
        onPress={onPress}
        testOnly_pressed={testOnly_pressed}>
        <View style={styles.iconContainer}>
          <Icon
            name={'checkcircle'}
            style={isActive ? styles.icon : styles.iconHidden}
          />
        </View>
        <Image source={category} style={styles.image} />
        <TextSmall text={text} fontColor={'black'} />
      </Pressable>
    </View>
  );
};
