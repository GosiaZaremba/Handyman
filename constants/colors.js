export const Colors = {
  button: {
    light: '#02a568',
    dark: '#71bc59',
  },
  gold: {
    light: '#ffaa63',
    dark: '#ff8e0b',
  },
  grey: {
    divider: '#d9d9d9',
    light: '#e5e5e5',
  },
  font: {
    green: '#44b35e',
    grey: '#9e9e9e',
    white: '#fff',
    darkGrey: '#515050',
  },
  icons: {
    facebook: '#3b5998',
    twitter: '#26a6d1',
    google: '#dd4b39',
  },
  icon: '#00b875',
  star: {
    grey: '#e1e1e1',
    gold: '#ffaa63',
  },
};
