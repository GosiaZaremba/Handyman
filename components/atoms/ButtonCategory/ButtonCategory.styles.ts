import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  container: {
    width: 120,
    height: 95,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.font.white,
    elevation: 8,
  },
  pressed: {
    width: 120,
    height: 95,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.button.categoryButton,
    borderWidth: 1,
    borderColor: Colors.icon,
  },
  pressable: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.font.white,
  },
  image: {
    width: 40,
    height: 40,
    zIndex: 1,
  },
  icon: {
    fontSize: 14,
    color: Colors.icon,
    marginRight: 7,
  },
  iconContainer: {
    height: 14,
    alignSelf: 'flex-end',
  },
  iconHidden: {
    fontSize: 15,
    color: Colors.icon,
    marginRight: 7,
    display: 'none',
  },
});
