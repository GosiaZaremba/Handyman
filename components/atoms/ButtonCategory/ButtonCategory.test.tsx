import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {ButtonCategory} from './ButtonCategory';
import {Colors} from '../../../constants/colors';

const mockOnPress = jest.fn();

describe('Atom ButtonCategory', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <ButtonCategory
          onPress={mockOnPress}
          testID="atom-buttonCategory"
          testOnly_pressed={true}
          category={require('../../../assets/png/05_icons_cleaning.png')}
          text={'Laundry'}
          isActive={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure onPress works correctly', () => {
    const {getByTestId} = render(
      <ButtonCategory
        onPress={mockOnPress}
        testID="atom-buttonCategory"
        testOnly_pressed={false}
        category={require('../../../assets/png/05_icons_cleaning.png')}
        text={'Laundry'}
        isActive={false}
      />,
    );
    const buttonCategory = getByTestId('atom-buttonCategory');
    fireEvent.press(buttonCategory);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
    expect(buttonCategory.props.style.backgroundColor).toBe(Colors.font.white);
  });

  test('ensure pressable pressed works correctly', () => {
    const {getByTestId} = render(
      <ButtonCategory
        onPress={mockOnPress}
        testID="atom-buttonCategory"
        testOnly_pressed={false}
        category={require('../../../assets/png/05_icons_cleaning.png')}
        text={'Laundry'}
        isActive={true}
      />,
    );
    const buttonCategory = getByTestId('atom-buttonCategory');
    fireEvent.press(buttonCategory);
    expect(buttonCategory.props.style.borderRadius).toBe(10);
    expect(buttonCategory.props.style.backgroundColor).toBe(
      Colors.button.categoryButton,
    );
  });
});
