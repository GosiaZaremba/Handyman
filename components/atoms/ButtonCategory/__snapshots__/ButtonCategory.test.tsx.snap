// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom ButtonCategory Component Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 8,
      "height": 95,
      "justifyContent": "center",
      "width": 120,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,184,117,.1)",
        "borderColor": "#00b875",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 95,
        "justifyContent": "center",
        "width": 120,
      }
    }
    testID="atom-buttonCategory"
  >
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "height": 14,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#00b875",
              "fontSize": 14,
              "marginRight": 7,
            },
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î˜°
      </Text>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../assets/png/05_icons_cleaning.png",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
          "zIndex": 1,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "OpenSans-Regular",
            "fontSize": 12,
            "lineHeight": 20,
            "textAlign": "center",
          },
          Object {
            "color": "black",
          },
        ]
      }
    >
      Laundry
    </Text>
  </View>
</View>
`;
